### PR TITLE
chore(tests): Add test for unsigned evm transaction instruction compatibility.

### PR DESCRIPTION
**Pull request by vldm**

### CODE REVIEW

File: evm-utils/programs/evm_loader/src/instructions.rs [hunk 1]
```diff
@@ -173,19 +174,152 @@ mod test {
         assert_eq!(&*data, &*custom_data)
     }
 
+    macro_rules! len_and_hex_buf{
```
Comment (hrls) [inline evm-utils/programs/evm_loader/src/instructions.rs]: fmt

File: evm-utils/programs/evm_loader/src/instructions.rs [hunk 2]
```diff
@@ -173,19 +174,152 @@ mod test {
...
+        ($buffer: expr, $data: ident $(,$fixed_len: expr)?) => {
```
Comment (hrls) [inline evm-utils/programs/evm_loader/src/instructions.rs:178]: `$(, $fixed_len: expr)`, but macro style is open question

File: evm-utils/programs/evm_loader/src/instructions.rs [hunk 3]
```diff
@@ -173,19 +174,152 @@ mod test {
...
+            {
+
+            let mut array_len: [u8; 8] = [66, 0, 0, 0, 0, 0, 0, 0];
+            let mut array_in_hex_bytes: [u8; 66] = [0; 66]; // max array len 2+32*2 = 0x + 2u8 for each byte
```
Comment (hrls) [inline evm-utils/programs/evm_loader/src/instructions.rs]: Check comment line, `0x` isn't typo?

File: evm-utils/programs/evm_loader/src/instructions.rs [hunk 4]
```diff
@@ -173,19 +174,152 @@ mod test {
...
+
+            let data_in_hex = format!("0x{:x}", $data);
```
Comment (hrls) [inline evm-utils/programs/evm_loader/src/instructions.rs]: `format!("{:#x}"), $data)`

File: evm-utils/programs/evm_loader/src/instructions.rs [hunk 5]
```diff
@@ -173,19 +174,152 @@ mod test {
...
+            len_and_hex_buf!(buffer, nonce);
+            len_and_hex_buf!(buffer, gas_price);
+            len_and_hex_buf!(buffer, gas_limit);
+
+            if let evm::TransactionAction::Call(receiver) = receiver {
+                let tag: [u8; 4] = [0, 0, 0, 0];
+                buffer.extend_from_slice(&tag);
+                len_and_hex_buf!(buffer, receiver, 20);
+            } else {
+                let tag: [u8; 4] = [1, 0, 0, 0];
```
Comment (hrls) [inline evm-utils/programs/evm_loader/src/instructions.rs]: I prefer obvious match arms. At least for reading purposes. `1` is `TransactionAction::Create`?

File: evm-utils/programs/evm_loader/src/instructions.rs [hunk 6]
```diff
@@ -173,19 +174,152 @@ mod test {
...
+
+            let mut array_len: [u8; 8] = [66, 0, 0, 0, 0, 0, 0, 0];
+            let mut array_in_hex_bytes: [u8; 66] = [0; 66]; // max array len 2+32*2 = 0x + 2u8 for each byte
+
+            let data_in_hex = format!("0x{:x}", $data);
+            assert!(data_in_hex.len() <= 0xff);
+            $(assert_eq!(data_in_hex.len(), $fixed_len * 2 + 2);)?
+
+            array_in_hex_bytes[0..data_in_hex.len()].copy_from_slice(data_in_hex.as_bytes());
+            // its not a valid number to little endian array encoding, but our len guaranted to be less < 255 bytes so its okay to write only first byte.
```
Comment (hrls) [inline evm-utils/programs/evm_loader/src/instructions.rs]: assert this

**Review by hrls — APPROVED** [review]

File: evm-utils/programs/evm_loader/src/instructions.rs (hunk 3 above)
Comment (vldm) [reply evm-utils/programs/evm_loader/src/instructions.rs]: No, 32 byte string represented as hex, wich gives us 64 bytes + "0x" in bytes as prefix, giveus 66.
(That how our current auto derived serialization for bincode works)

**Review by hrls — APPROVED** [review]

**Pull request merged**